File: app/src/main/java/es/santirivera/surveilfall/di/modules/ActivityModule.java
package es.santirivera.surveilfall.di.modules;


import dagger.Module;
import dagger.Provides;
import es.santirivera.surveilfall.base.activity.BaseActivity;

@Module
public class ActivityModule {

    private final BaseActivity activity;

    public ActivityModule(BaseActivity activity) {
        this.activity = activity;
    }

    @Provides
    BaseActivity provideActivity() {
        return this.activity;
    }

}

File: app/src/main/java/es/santirivera/surveilfall/di/modules/ActivityModule.kt
package es.santirivera.surveilfall.di.modules


import dagger.Module
import dagger.Provides
import es.santirivera.surveilfall.base.activity.BaseActivity

@Module
class ActivityModule(private val activity: BaseActivity) {

    @Provides
    internal fun provideActivity(): BaseActivity {
        return this.activity
    }

}